### PR TITLE
Implement decl_attribute (fixes #22)

### LINKED ISSUE #22: Add a macro for procedural macro attributes
https://doc.rust-lang.org/reference/procedural-macros.html#attribute-macros

These would probably benefit from a helper macro like `decl_derive`.

### PR DESCRIPTION
**Pull request by nox**

### CODE REVIEW

**Comment by mystor** [comment]
It would be nice to also have testing attribute support for the new attribute, but this looks pretty good, thanks!

File: src/macros.rs [hunk 1]
```diff
@@ -72,6 +72,62 @@ macro_rules! decl_derive {
...
+/// }
+///
+/// # const _IGNORE: &'static str = stringify! {
+/// decl_attribute!([interesting] => attribute_interesting);
+/// # };
+/// ```
+/// ```
+#[macro_export]
+macro_rules! decl_attribute {
+    // XXX: Switch to using this variant everywhere?
```
Comment (mystor) [inline src/macros.rs]: Im not sure what you're referring to with this comment?

Comment (nox) [reply src/macros.rs]: Leftover from copy-pasting `decl_derive`.

**Comment by mystor** [comment]
Sorry about how long this took to merge, somehow this PR got lost in my notifications.

**Pull request merged**